pkg/controller/logcollector/logcollector_controller.go

```diff
@@ -181,7 +181,7 @@ func (r *ReconcileLogCollector) Reconcile(ctx context.Context, request reconcile
 		return reconcile.Result{}, nil
 	}
 
-	if err = utils.CheckLicenseKey(ctx, r.client); err != nil {
+	if _, err := utils.FetchLicenseKey(ctx, r.client); err != nil {
```

Comment:
If there is an error, we cannot conclude it is because the license was not found. It could be any error. Therefore I think we should improved the degraded message to something like: "Unable to fetch License".
There are multiple occurrences of this in the PR.

Reply:
We can check the specific type of error - we do it elsewhere.

i.e., if it's a "NotFound" error it's different than a connection error. We should probably do that here.

Reply:
@rene-dekker I actually see that we set `r.status.SetDegraded("License not found", err.Error())`

This means that on the status of the tigera status we set something like. 
  - lastTransitionTime: "2021-02-10T19:16:11Z"
    message: License not found
    reason: (ks8 API timeout | license not found)
    status: "True"

Reply:
In other places we change the message as well, based on the type of error. I'm not super picky about it, but for consistency it might be better.